internal/mode/static/usage/reporter.go

```diff
@@ -0,0 +1,128 @@
+type PodDetails struct {
+	// CurrentPodsCount is the total count of NGF NGINX Plus Pods in the cluster.
+	CurrentPodCounts CurrentPodsCount `json:"current_pod_counts"`
+}
+
+// CurrentPodsCount is the total count of NGF NGINX Plus Pods in the cluster.
+type CurrentPodsCount struct {
+	// PodCount is the current count of NGF NGINX Plus Pods in the cluster.
+	PodCount int64 `json:"pod_count"`
+	// DosCount is the count of Pods with NAP DOS enabled in the cluster. Not applicable for NGF,
+	// but required as part of the payload.
+	DosCount int64 `json:"dos_count"`
+	// WafCount is the count of Pods with NAP WAF enabled in the cluster. Not applicable for NGF,
+	// but required as part of the payload.
+	WafCount int64 `json:"waf_count"`
+}
+
+// Reporter reports the NGINX Plus usage info to the provided collector.
+type Reporter struct {
+	// secret contains the credentials for the usage collector.
+	secret SecretStore
+	// endpoint is the API endpoint of the usage collector.
+	endpoint string
+	// insecureSkipVerify controls whether the client verifies the server cert. Used in testing.
+	insecureSkipVerify bool
+}
+
+// NewReporter creates a new usage reporter.
+func NewReporter(
+	secret SecretStore,
+	endpoint string,
+	insecureSkipVerify bool,
+) *Reporter {
+	return &Reporter{
+		secret:             secret,
+		endpoint:           endpoint,
+		insecureSkipVerify: insecureSkipVerify,
+	}
+}
+
+// Report sends a PUT request with the provided data to the API endpoint configured in the Reporter.
+// The clusterUID is used as the name in the API path.
+func (r *Reporter) Report(ctx context.Context, data ClusterDetails) error {
+	buf, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	endpointURL, err := url.Parse(r.endpoint)
+	if err != nil {
+		return err
+	}
+
+	queryURL := endpointURL.JoinPath(data.Metadata.UID)
+	bodyReader := bytes.NewReader(buf)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, queryURL.String(), bodyReader)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	username, password := r.secret.GetCredentials()
+	if username == nil || password == nil {
+		return errors.New("username or password not set for NGINX Plus usage reporting; unable to send reports. " +
+			"Ensure that the usage Secret exists and the username and password are set.")
+	}
+	req.SetBasicAuth(string(username), string(password))
+
+	client := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: r.insecureSkipVerify, //nolint:gosec // used for testing
+			},
+		},
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read the response body: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("non-200 response: %v\nresponse body: %v", resp.StatusCode, string(body))
```

Comment:
would error messages with new line characters look ok in the logs?

Reply:
good point, I can remove that